Source/Xamarin/Prism.Forms/Services/PageDialogService/ActionSheetButton.cs

```diff
@@ -1,74 +1,201 @@
-﻿using System.Windows.Input;
+﻿using System;
+using System.Windows.Input;
 
 namespace Prism.Services
 {
     /// <summary>
     /// Represents a button displayed in <see cref="Prism.Services.IPageDialogService.DisplayActionSheetAsync(string, IActionSheetButton[])"/>
     /// </summary>
-    public class ActionSheetButton : IActionSheetButton
+    public class ActionSheetButton : ActionSheetButtonBase
     {
+        protected internal ActionSheetButton() { }
+
         /// <summary>
-        /// The button will be used as destroy
+        /// Action to perform when the button is pressed
         /// </summary>
-        public virtual bool IsDestroy { get; protected set; }
+        /// <value>The action.</value>
+        public Action Action { get; protected set; }
+
+        /// <inheritDoc />
+        protected override void OnButtonPressed()
+        {
+            Action?.Invoke();
+
+            if( Command?.CanExecute( null ) ?? false )
+                Command.Execute( null );
+        }
+
+        #region Factory Methods
```

Comment:
@brianlagunas shouldn't we have an ActionSheetButtonFactory instead of hiding this in here?